src/main/java/crazypants/enderio/conduit/gui/item/ItemSettings.java

```diff
@@ -212,9 +214,13 @@ public void actionPerformed(@Nonnull GuiButton guiButton) {
         } else if (guiButton.id == ID_EXTRACT_CHANNEL) {
             DyeColor col = DyeColor.fromIndex(extractChannelB.getColorIndex());
             itemConduit.setInputColor(gui.getDir(), col);
+        } else {
+            needUpdate = false;
+            if (insertFilterGui != null) insertFilterGui.actionPerformed(guiButton);
+            if (extractFilterGui != null) extractFilterGui.actionPerformed(guiButton);
```

Comment:
Now that this is in `else` you can just return instead of setting the boolean. And yeah this looks fine to me

Reply:
fixed